src/test/java/seedu/jarvis/ui/MainWindowTest.java

```diff
@@ -4,23 +4,43 @@
 
 import org.junit.jupiter.api.BeforeAll;
 
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.io.TempDir;
+import seedu.jarvis.logic.Logic;
+import seedu.jarvis.logic.LogicManager;
 import seedu.jarvis.model.Model;
 import seedu.jarvis.model.ModelManager;
+import seedu.jarvis.storage.JsonAddressBookStorage;
+import seedu.jarvis.storage.JsonUserLoginStorage;
+import seedu.jarvis.storage.JsonUserPrefsStorage;
+import seedu.jarvis.storage.StorageManager;
 import seedu.jarvis.testutil.TypicalManagers;
 
+import java.nio.file.Path;
+
 public class MainWindowTest {
 
     // Integration and unit tests combined
 
+    @TempDir
+    public Path temporaryFolder;
+
     // This model is used to execute commands
     private Model model;
 
+    private Logic logic;
+
     @BeforeAll
     public void setUp() {
```

Comment:
Thank you for writing this, i will use it to write integration tests for MainWindow, LogicManager and the other classes involved from methods called in MainWindow.java